doc/source/whatsnew/v2.0.0.rst

```diff
@@ -291,6 +291,39 @@ The new behavior, as for datetime64, either gives exactly the requested dtype or
    ser.astype("timedelta64[s]")
    ser.astype("timedelta64[D]")
 
+.. _whatsnew_200.api_breaking.zero_len_indexes:
+
+Empty DataFrames/Series will now default to have a ``RangeIndex``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When users don't specify indexes or columns when instantiating :class:`DataFrame` or :class:`Series`
+objects, pandas infers the index type for them. Previously, if it was not possible to
+infer the index type from the data, pandas used to infer the index to be a :class:`RangeIndex`,
+except when the frame/series was empty, then the index was inferred to be an :class:`Index`
+with dtype ``object``.
+
+This has been changed, so the type of index/columns will always be :class:`RangeIndex`,
+when users don't declare the index and it's not possible to infer the index type from the data (:issue:`49572`).
```

Comment:
> infer the index type from the data

Does this refer to the `data` argument in `Series(data=..., )`, `DataFrame(data=...)`?

Reply:
Yes. It is maybe unclear if the paragraph points to the index data?

Reply:
Yeah generally, these paragraphs were a little unclear to me

Is it fair to say:

```
Before, constructing an empty (length 0) Series or DataFrame without specifying the axes (index=None, columns=None) would return the axes as empty Indexes with object dtype. Now, the axes return an empty RangeIndex 
``` 

?

Reply:
👍 . I can reword it. I’ll do it tomorrow, it’s quite late here now. Same with the other issue.